setup.py

```diff
@@ -37,7 +37,6 @@
 
 
 requirements = [
-    'attr',
```

Comment:
Should add `attrs` as a dev requirement along with `pytest`.

Reply:
It all depends on how strict, and verbose, you want your dev deps to be. `pytest` already specifcies that it [requires `attrs`](https://github.com/pytest-dev/pytest/blob/main/setup.cfg#L44) to install.

If we include `attrs` do we also include the rest of the `pytest` deps?

Reply:
Well, since we directly import `attrs` in the tests I think specifying it explicitly in `setup.py` makes sense. We don't import any of the other pytest deps, so no need to include those. This follows the "include what you use" principle, which is more of a C++ thing but generally good practice.

Reply:
Works for me! I've added it to `extra_requires` in `setup.py`, right along side `pytest`.